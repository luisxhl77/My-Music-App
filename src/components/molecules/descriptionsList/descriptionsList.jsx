import PropTypes from 'prop-types';
import imageFavorite from '../../../img/favorite.png';
import './descriptionsList.scss';

export const DescriptionsList = ({ name, images, owner, followers, tracks }) => {

    return (
        <section className="DescriptionsList">
            <figure className="DescriptionsList__image">
                <img src={ images } alt="" className="DescriptionsList__img"/>
            </figure>
            <article className="DescriptionsList__infoText">
                <h5> LIST </h5>
                <p className="DescriptionsList__name-list">
                    { name }
                </p>
                <ul className="DescriptionsList__descrip-list">
                    <li className="DescriptionsList__owner"> { owner } - </li>
                    <li className="DescriptionsList__list"> { followers } " I like it " </li>
                    <li className="DescriptionsList__list"> { tracks } songs </li>
                </ul>
            </article>
        </section>
    )

}

DescriptionsList.propTypes = {
    name: PropTypes.string.isRequired,
    images: PropTypes.string,
    owner: PropTypes.string,
    followers: PropTypes.number,
    tracks: PropTypes.number,
}

DescriptionsList.defaultProps = {
    name: 'no title',
    images: imageFavorite,
    owner: 'no owner',
    followers: 0,
    tracks: 0,
}
