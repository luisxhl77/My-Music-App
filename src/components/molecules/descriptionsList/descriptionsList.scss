@import '../../../styles/variables.scss';
@import '../../../styles/mixins.scss';
.DescriptionsList{
    @include flexbox(none, none, none, none, flex-end);
    padding: 0px 1% 0px 1%;
    &__infoText{
        flex: 1;
        & h1{
            margin-bottom: 10px;
        }
    }
    &__image{
        width: 20vw;
        margin: 7vh 25px 0px 25px;
        box-shadow: 2px 2px 30px #000;
    }
    &__name-list{
        margin: 0px 0px 26px 0px;
        font-size: 4.9vw;
        font-weight: 900;
        text-transform: lowercase;
        letter-spacing: -1px;
        word-spacing: -4px;
        &:first-letter {
            text-transform: uppercase;
        }
    }
    &__descrip-list{
        color: #ffffff;
    }
    &__owner{
        float: left;
        margin-right: 5px;
        font-weight: 700;
        font-size: 13px;
    }
    &__list{
        float: left;
        margin-right: 5px;
        font-size: 13px;
    }
}
.MuiAvatar-root{
    width: 10px;
}