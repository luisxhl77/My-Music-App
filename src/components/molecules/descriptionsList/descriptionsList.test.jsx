import { render, screen } from "@testing-library/react"
import { DescriptionsList } from "./descriptionsList"

describe('Pruebas de <DescriptionList/>', () => {
    const name = 'Wake Me Up'; 
    const owner = 'Avicii'; 
    const images = 'https://avicii.com/avicii.jsg'; 
    const tracks = 10;
    const followers = 180; 

    test('Debe de hacer match con el snapshot', () =>{

        const { container } = render( <DescriptionsList 
            name      = { name }
            owner     = { owner }
            images    = { images }
            tracks    = { tracks }
            followers = { followers }
        />);
        expect( container ).toMatchSnapshot();
    
    })

    test('debe de mostrar src de la imagen', () => {
   
        render( <DescriptionsList 
            name      = { name }
            owner     = { owner }
            images    = { images }
            tracks    = { tracks }
            followers = { followers }
        />);

        const { src } = screen.getByLabelText('imageDescription');
        expect( src ).toBe( images );

    })

    test('debe de mostrar el name, owner, tracks y followers', () => {
   
        render( <DescriptionsList 
            name      = { name }
            owner     = { owner }
            images    = { images }
            tracks    = { tracks }
            followers = { followers }
        />);

        expect( screen.getByText( name  ) ).toBe( screen.getByLabelText('nameList'));
        expect( screen.getByText( owner ) ).toBe( screen.getByLabelText('ownerList'));
        expect( screen.getByText( tracks ) ).toBe( screen.getByLabelText('tracksList'));
        expect( screen.getByText( followers ) ).toBe( screen.getByLabelText('followersList'));
    })

})