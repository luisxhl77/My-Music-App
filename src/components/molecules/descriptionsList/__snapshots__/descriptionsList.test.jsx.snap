// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <DescriptionList/> Debe de hacer match con el snapshot 1`] = `
<div>
  <section
    class="DescriptionsList"
  >
    <figure
      class="DescriptionsList__image"
    >
      <img
        alt="coverList"
        aria-label="imageDescription"
        class="DescriptionsList__img"
        src="https://avicii.com/avicii.jsg"
      />
    </figure>
    <article
      class="DescriptionsList__infoText"
    >
      <h5>
         LIST 
      </h5>
      <p
        aria-label="nameList"
        class="DescriptionsList__name-list"
      >
        Wake Me Up
      </p>
      <ul
        class="DescriptionsList__descrip-list"
      >
        <li
          aria-label="ownerList"
          class="DescriptionsList__owner"
        >
          Avicii
        </li>
        <li
          aria-label="followersList"
          class="DescriptionsList__list"
        >
          180
        </li>
        <li
          class="DescriptionsList__list"
        >
          " I like it "
        </li>
        <li
          aria-label="tracksList"
          class="DescriptionsList__list"
        >
          10
        </li>
        <li
          class="DescriptionsList__list"
        >
          songs
        </li>
      </ul>
    </article>
  </section>
</div>
`;
